### PR TITLE
Support xpu timer metric collection.

### PR DESCRIPTION
**Pull request by BalaBalaYi**
### What changes were proposed in this pull request?

1. Use a async thread to collect metric periodically.
2. Pre process and report the metric to master.
3. Refactor the implementation of 'diagnosis_data'.
4. Other optimization.

### Why are the changes needed?

To support training hang detection later.

### Does this PR introduce any user-facing change?

No.

### How was this patch tested?

UT.

### CODE REVIEW

File: dlrover/python/diagnosis/common/diagnosis_data.py [hunk 1]
```diff
@@ -11,69 +11,117 @@
...
+        return self._node_rank
+
+    def is_resolvable(self):
+        if self.data_type == DiagnosisDataType.TRAINING_HANG_DETECTION:
+            return True
+        # TODO: add more resolvable metric type later
+        return False
+
+
+class TrainingLog(AgentMetric):
```
Comment (samplise) [inline dlrover/python/diagnosis/common/diagnosis_data.py:105]: I am a little confused here. Why do AgentMetrics inherit DiagnosisData and TrainingLog inherit AgentMetrics? I fail to catch up with the design of AgentMetrics here.

File: dlrover/python/diagnosis/common/constants.py [hunk 2]
```diff
@@ -12,12 +12,27 @@
...
-class DiagnoseAction(object):
+class DiagnosisConstant(object):
+    MASTER_DIAGNOSIS_OBSERVING_INTERVAL_SECS = 180
+    AGENT_PERIODICALLY_DIAGNOSIS_INTERVAL_SECS = 60
+
+
+class DiagnosisDataType(object):
+    GENERIC = "GENERIC"
+    TRAINING_LOG = "TRAINING_LOG"
+    TRAINING_HANG_DETECTION = "TRAINING_HANG_DETECTION"
```
Comment (samplise) [inline dlrover/python/diagnosis/common/constants.py:32]: "TRAINING_HANG_DETECTION" seems not to be the type name of data. Maybe stack data or something similar?

File: dlrover/python/diagnosis/datacollector/metrics_collector.py [hunk 3]
```diff
@@ -11,20 +11,18 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from dlrover.python.diagnosis.common.diagnosis_data import ChipMetrics
 from dlrover.python.diagnosis.datacollector.data_collector import DataCollector
 
 
 class MetricsCollector(DataCollector):
```
Comment (samplise) [inline dlrover/python/diagnosis/datacollector/metrics_collector.py:17]: I think we can remove metrics collector directly and make XPU timer collector inherit data collector directly.

File: dlrover/python/common/grpc.py [hunk 4]
```diff
@@ -490,17 +490,19 @@ class NodeCheckpointState(Message):
 
 @dataclass
 class DiagnosisTrainingLog(Message):
```
Comment (BalaBalaYi) [inline dlrover/python/common/grpc.py:492]: Need removed.

File: dlrover/python/diagnosis/common/diagnosis_data.py (hunk 1 above)
Comment (BalaBalaYi) [reply dlrover/python/diagnosis/common/diagnosis_data.py]: Refactored.

File: dlrover/python/diagnosis/common/constants.py (hunk 2 above)
Comment (BalaBalaYi) [reply dlrover/python/diagnosis/common/constants.py]: Change to ```XPU_TIMER_METRIC```

File: dlrover/python/diagnosis/datacollector/metrics_collector.py (hunk 3 above)
Comment (BalaBalaYi) [reply dlrover/python/diagnosis/datacollector/metrics_collector.py]: Done.

**Review by samplise — APPROVED** [review]
lgtm

**Pull request merged**